packages/tax/src/models/tax-rate-rule.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Cascade,
+  Entity,
+  ManyToOne,
+  PrimaryKey,
+  PrimaryKeyProp,
+  Property,
+} from "@mikro-orm/core"
+import TaxRate from "./tax-rate"
+
+const TABLE_NAME = "tax_rate_rule"
+
+const taxRateIdIndexName = "IDX_tax_rate_rule_tax_rate_id"
+
+@Entity({ tableName: TABLE_NAME })
+export default class TaxRateRule {
+  @PrimaryKey({ columnType: "text" })
+  tax_rate_id!: string
+
+  @PrimaryKey({ columnType: "text" })
+  reference_id!: string;
+
+  [PrimaryKeyProp]?: ["tax_rate_id", "reference_id"]
+
+  @Property({ columnType: "text" })
+  reference_type: string
```

Comment:
Depending on how filters will be passed to fetch data from this table, probably individual indexes on tax_rate_id, reference_id and reference_type will be a better choice than this composite primary key.
Keeping the composite key and adding additional indexes is also an option.